tests/compat.sh

```diff
@@ -1332,20 +1331,19 @@ SKIP_NEXT="NO"
 trap cleanup INT TERM HUP
 
 for MODE in $MODES; do
-    PSK_TESTS=""
-    for VERIFY in $VERIFIES; do
-        VERIF=$(echo $VERIFY | tr '[:upper:]' '[:lower:]')
-        for TYPE in $TYPES; do
-
-            if [ "$TYPE" = "PSK" ]; then
-                if [ -z "$PSK_TESTS" ]; then
-                    PSK_TESTS="FINISHED"
-                    VERIF="no"
-                else
-                    continue;
-                fi
-            fi
+    for TYPE in $TYPES; do
+
+        # PSK cipher suites do not allow client certificate verification.
+        # This means PSK test cases with VERIFY=YES should be replaced by
+        # VERIFY=NO or be ignored. SUB_VERIFIES variable is used to constrain
+        # verification option for PSK test cases.
+        SUB_VERIFIES=$VERIFIES
+        if [ "$TYPE" = "PSK" ]; then
+            SUB_VERIFIES="NO"
+        fi
 
+        for VERIFY in $SUB_VERIFIES; do
+            VERIF=$(echo $VERIFY | tr '[:upper:]' '[:lower:]')
```

Comment:
Reminder: since we didn't backport #6902, see [comment](https://github.com/Mbed-TLS/mbedtls/pull/6902#issuecomment-1415661641), this PR is little bit different from #7052.
This line of code is moved from [line 1137](https://github.com/Mbed-TLS/mbedtls/pull/7070/commits/21daf2930a343068e6d318d0c827839e9b276cfd#diff-5eb12b5f615f14f9cdb176866c7218f2fe97c7c9fe3b836bce0f13afab858be9L1137). It doesn't impact `compat.sh` so this change is not removed. Let me know if it's out of scope in this PR.